agent/consul/gateways/gateway_meta.go

```diff
@@ -137,3 +181,118 @@ func (g *gatewayMeta) boundListenerByName(name string) (int, *structs.BoundAPIGa
 	}
 	return -1, nil
 }
+
+func (g *gatewayMeta) checkCertificates(store *state.Store) (map[structs.ResourceReference]error, error) {
+	certificateErrors := map[structs.ResourceReference]error{}
+	for i, listener := range g.Gateway.Listeners {
+		bound := g.BoundGateway.Listeners[i]
+		for _, ref := range listener.TLS.Certificates {
+			_, certificate, err := store.ConfigEntry(nil, ref.Kind, ref.Name, &ref.EnterpriseMeta)
+			if err != nil {
+				return nil, err
+			}
+			if certificate == nil {
+				certificateErrors[ref] = errors.New("certificate not found")
+			} else {
+				bound.Certificates = append(bound.Certificates, ref)
+			}
+		}
+	}
+	return certificateErrors, nil
+}
+
+func (g *gatewayMeta) checkConflicts() (structs.ControlledConfigEntry, bool) {
+	now := time.Now()
```

Comment:
Similar with the UTC question above. I think if we always parse back into a Go code `Time` object, it should be fine. It's a bit of an edge case. 🤷🏻‍♂️

Reply:
I'll swap all of these to use UTC -- which was a good catch. This is only used for users to see when a status condition was set though, so it shouldn't be problematic for any other code.